File: codesignal/arcade/2.core/6.labyrinth of nested loops/1.isPower.cpp
// Determine if the given number is a power of some non-negative integer.

#include <cmath>
using namespace std;

class Solution {
public:
  bool isPower(int n) {
    if (n == 1) return true;
    for (int i = 2; i < n; i++) {
      int j = 2;
      while (pow(i, j) <= n) {
        if (pow(i, j) == n) return true;
        j++;
      }
    }
    return false;
  }
};

File: codesignal/arcade/2.core/6.labyrinth of nested loops/1.isPower.py
# Determine if the given number is a power of some non-negative integer.

class Solution:
    def isPower(self, n):
        if n == 1:
            return True
        for i in range(2, n):
            for j in range(2, n):
                if i ** j == n:
                    return True
        return False